pkg/apiserver/domain/service/application.go

```diff
@@ -1844,23 +1851,39 @@ func (c *applicationServiceImpl) handlePolicyBindingWorkflowStep(ctx context.Con
 
 // checkPolicyUsedByWorkflow check a policy whether used by any workflow step
 func (c *applicationServiceImpl) checkPolicyUsedByWorkflow(ctx context.Context, app *model.Application, policyName string) (bool, error) {
-	workflows, err := c.WorkflowService.ListApplicationWorkflow(ctx, app)
+	wlb, err := c.findAllBindingPolicyWorkflowStep(ctx, app, policyName)
 	if err != nil {
 		return false, err
 	}
+	return len(wlb) != 0, nil
+}
+
+func (c *applicationServiceImpl) findAllBindingPolicyWorkflowStep(ctx context.Context, app *model.Application, policyName string) ([]apisv1.WorkflowPolicyBinding, error) {
+	var res []apisv1.WorkflowPolicyBinding
+	workflows, err := c.WorkflowService.ListApplicationWorkflow(ctx, app)
+	if err != nil {
+		return nil, err
+	}
 	for _, w := range workflows {
+		var wlb apisv1.WorkflowPolicyBinding
+		have := false
 		for _, step := range w.Steps {
 			p := step.Properties
 			policies, _, err := extractPolicyListAndProperty(p)
 			if err != nil {
-				return false, err
+				return nil, err
 			}
 			for _, policy := range policies {
 				if policy == policyName {
-					return true, nil
+					wlb.Steps = append(wlb.Steps, step.Name)
+					have = true
```

Comment:
If  policy is found, break the iterator.

Reply:
Cannot break, should find all workflow and workflowStep